python/ngen_cal/src/ngen/cal/optimizers/grey_wolf.py

```diff
@@ -0,0 +1,404 @@
+from collections import namedtuple, deque
+import logging
+import numpy as np
+from pathlib import Path
+import pandas as pd
+import multiprocessing as mp
+from pyswarms.base import SwarmOptimizer
+from pyswarms.backend import Swarm, generate_swarm
+from pyswarms.backend.topology import Star
+from pyswarms.utils.reporter import Reporter
+from pyswarms.backend.operators import compute_pbest, compute_objective_function
+
+import shutil
+import time
+from typing import Tuple, List, Optional
+
+def create_swarm(
+    n_particles,
+    dimensions,
+    bounds=None,
+    center=1.0,
+    init_pos=None,
+    options={}
+):
+    """Generate a Swarm class with no velocity
+
+    The pyswarms backend uses a velocity component for general particle
+    swarms.  The Grey Wolf doesn't use this, and needs to create
+    a swarm with no velocity, so this function passes an empty array for velocities
+    to the base Swarm.
+
+    see :func:`pyswarms.backend.create_swarm` for argument descriptions
+    """
+    position = generate_swarm(
+        n_particles,
+        dimensions,
+        bounds=bounds,
+        center=center,
+        init_pos=init_pos,
+    )
+
+    return Swarm(position, np.array([]), options=options)
+
+
+class GreyWolfOptimizer(SwarmOptimizer):
+    """_summary_
+
+    Args:
+        SwarmOptimizer (_type_): _description_
```

Comment:
Nope, this is just @hellkite500's IDE defaults. @hellkite500, can you add docs for this?

Reply:
A bit of an aside:  I originally started my review for `grey_wolf.py` before it got moved to the private subpackage, but didn't finish before it got moved, and Github made me refresh and, seemingly, required I start over in the "other" `grey_wolf.py`.  So some of my comments may appear to be duplicated.